src/open_company/resources/common.clj

```diff
@@ -143,6 +144,22 @@
     (select-keys [:avatar :image :user-id :name])
     (clojure.set/rename-keys {:avatar :image})))
 
+;; ----- DB Access Timeouts ----
+
+(def default-timeout 1000) ; 1 sec
+
+(defmacro with-timeout
+  "A basic macro to wrap things in a timeout.
+  Will throw an exception if the operation times out.
+  Note: This is a simplistic approach and piggiebacks on core.asyncs executor-pool.
+  Read this discussion for more info: https://gist.github.com/martinklepsch/0caf92b5e42eefa3a894"
+  [ms & body]
+  `(let [c# (async/thread-call #(do ~@body))]
+     (let [[v# ch#] (async/alts!! [c# (async/timeout ~ms)])]
+       (if-not (= ch# c#)
+         (throw (ex-info "Operation timed out" {}))
```

Comment:
Isn't the idea @martinklepsch to use invalidate here to get that connection dumped from the pool? Seems this will prevent the long term lockups, but we'll still keep getting the bad connection from the pool.